src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntryHelper.js

```diff
@@ -257,15 +257,15 @@
 
     /**
      * @description: set new total amount and count variables and update data table after dry run completes
-     * @param updatedRecord: DataImport__c record that needs to be updated in the table
-     * @param isNewRecord: flag to indicate if record is new or existing
+     * @param model: model with records that need to be updated in the table
+     * @param isNewRecord: flag to indicate if newly entered record
      */
     afterDryRun: function(component, model, isNewRecord) {
         this.setTotals(component, model);
         if (model.dataImportRows.length > 0) {
-            //we only process one record at a time, but the model passes back a list
-            let newRecord = model.dataImportRows[0];
-            this.updateDataTableAfterDryRun(component, newRecord, isNewRecord);
+            for (let i = 0; i < model.dataImportRows.length; i++) {
+                this.updateDataTableAfterDryRun(component, model.dataImportRows[i], isNewRecord);
```

Comment:
For the instances where a single row is being changed or added, does the model return a list size 1 in model.dataImportRows? Just want to make sure we aren't iterating over the entire table in the cases where only a single row is being modified

Reply:
Longer-term thought, but the "updateDataTableAfterDryRun" is doing a lot of moving row data around / flattening that might be avoidable. I imagine we had a compelling reason to set up the data that way originally - I'm curious if @bethbrains has thoughts about this. I wouldn't necessarily pursue that change for this sprint though unless we feel like we have it really well t'd up

Reply:
Can confirm that only a single record at a time is actually inside model.dataImportRows when adding a new line or updating records inline

Reply:
Thanks @psadie - I'm fine with this as it stands then, thank you for clarifying